### PR TITLE
351 handle concurrent record updates record locking

### PR DESCRIPTION
**Pull request by rik86future**
You can use incognito sessions for testing this side by side.

**Scenario 1: Change need**
User 1 views a need (eg /need/1)
User 2 views the same need (eg /need/1)
User 1 changes status to Complete
User 2 changes something else
User 1 see Success
User 2 see Failure

**Scenario 2: Change person**
User 1 view the edit person page (eg /person/1/edit)
User 3 view the same edit person page (eg /person/1/edit)
User 1 changes Name
User 2 changes the Name or anything else
User 1 see Success
User 2 see Failure

**Scenario 3: Change person from triage**
User 1 view the edit person page (eg /person/1/edit)
User 3 view the same person triage page (eg /triage/1/edit)
User 1 changes Name
User 2 changes the Name or anything else
User 1 see Success
User 2 see Failure

**Additionally**:
Real time notifications will pop up if two users are accessing the same Person record in either triage mode or edit mode. The notifications refresh at a 3 seconds rate approx

### CODE REVIEW

**Comment by rik86future** [comment]
[Handle concurrent record updates / Record Locking](https://trello.com/c/pftKsDsG/351-handle-concurrent-record-updates-record-locking)

File: features/step_definitions/edit_need_steps.rb [hunk 1]
```diff
@@ -48,6 +48,27 @@
   expect(assignee_column).to have_content @expected_assignee
 end
 
+When("I change someone else's need status to 'complete'") do
+  visit "/needs/#{@need.id}"
+
+  Capybara.using_session('Second_users_session') do
```
Comment (jamiebuckley) [inline features/step_definitions/edit_need_steps.rb:54]: nice!

Comment (rik86future) [reply features/step_definitions/edit_need_steps.rb:54]: I learnt from the master ;)

**Comment by jamiebuckley** [comment]
This seems to work well, can't break it.  Nice job

**Pull request merged**